src/components/Tool/FixPhoneNumbers/FixPhoneNumbers.tsx

```diff
@@ -131,137 +99,22 @@ const FixPhoneNumbers: React.FC<Props> = ({ accountListId }: Props) => {
     const existingSources = new Set<string>();
     existingSources.add(manualSourceValue);
 
-    const newDataState = data
-      ? data.people.nodes?.reduce(
-          (map, person) => ({
-            ...map,
-            [person.id]: {
-              phoneNumbers: person.phoneNumbers.nodes.map((phoneNumber) => {
-                existingSources.add(phoneNumber.source);
-                return { ...phoneNumber };
-              }),
-            },
-          }),
-          {},
-        )
-      : {};
-    setDataState(newDataState);
+    if (data?.people?.nodes) {
+      data.people.nodes.forEach((person) => {
+        person.phoneNumbers.nodes.forEach((phoneNumber) => {
+          existingSources.add(phoneNumber.source);
+        });
+      });
+    }
     setSourceOptions([...existingSources]);
   }, [data]);
 
   const handleSourceChange = (event: SelectChangeEvent): void => {
     setDefaultSource(event.target.value);
   };
 
-  const handleChange = (
-    personId: string,
-    numberIndex: number,
-    newNumber: string,
-  ): void => {
-    const temp = { ...dataState };
-    dataState[personId].phoneNumbers[numberIndex].number = newNumber;
-    setDataState(temp);
-  };
-
   const handleBulkConfirm = async () => {
```

Comment:
This handler doesn't need to be `async` anymore.

```suggestion
  const handleBulkConfirm = () => {
```